### DISCUSSION TITLE
unable to load module '/usr/local/Cellar/imagemagick/7.0.10-29/lib/ImageMagick/modules-Q16HDRI/coders/png.la'

### DISCUSSION THREAD

**Discussion by haoshuai0229** [ImageMagick/ImageMagick#2598] [Help] [answered]
find . -iname "*.png" -exec echo {} \; -exec convert {} {} \;
./Resources/graphics/png/button-score-on.png
convert: unable to load module '/usr/local/Cellar/imagemagick/7.0.10-29/lib/ImageMagick/modules-Q16HDRI/coders/png.la': file not found @ error/module.c/OpenModule/1287.
convert: no decode delegate for this image format `PNG' @ error/constitute.c/ReadImage/562.
convert: no images defined `./Resources/graphics/png/button-score-on.png' @ error/convert.c/ConvertImageCommand/3285.
./Resources/graphics/png/button-score-off@2x.png

What I've Tried:

- Brew reinstall ImageMagick

- brew remove imagemagick
   brew install imagemagick --build-from-source

- stackover flow

- google search 

Please help me , thank you very much~

**Comment by snibgo** [comment]
What ImageMagick command did you try?

As you have IM v7, I suggest you use "magick", not "convert".

> no decode delegate for this image format PNG

This suggests your build does not include the internal PNG delegate.

What is the text outout from "magick -version"? And from "magick -list format"?

**Reply by haoshuai0229** [reply]
I want change all png image's hash , so i use this command. 
`find . -iname "*.png" -exec echo {} ; -exec convert {} {} ;`

This is my ImageMagick version
`magick -version
Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-18 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1)
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib`



magick -list format
Format  Module    Mode  Description
-------------------------------------------------------------------------------
    3FR  DNG       r--   Hasselblad CFV/H3D39II
    3G2  MPEG      r--   Media Container
    3GP  MPEG      r--   Media Container
    AAI* AAI       rw+   AAI Dune image
    ART* ART       rw-   PFS: 1st Publisher Clip Art
    ARW  DNG       r--   Sony Alpha Raw Image Format
    ASHLAR* ASHLAR    -w+   Image sequence laid out in continuous irregular courses
    AVI  MPEG      r--   Microsoft Audio/Visual Interleaved
    AVIF* HEIC      rw+   AV1 Image File Format (1.8.0)
    AVS* AVS       rw+   AVS X image
    BGR* BGR       rw+   Raw blue, green, and red samples
    BGRA* BGR       rw+   Raw blue, green, red, and alpha samples
    BGRO* BGR       rw+   Raw blue, green, red, and opacity samples
    BMP* BMP       rw-   Microsoft Windows bitmap image
    BMP2* BMP       rw-   Microsoft Windows bitmap image (V2)
    BMP3* BMP       rw-   Microsoft Windows bitmap image (V3)
    BRF* BRAILLE   -w-   BRF ASCII Braille format
    CAL* CALS      rw-   Continuous Acquisition and Life-cycle Support Type 1 Specified in MIL-R-28002 and MIL-PRF-28002
    CALS* CALS      rw-   Continuous Acquisition and Life-cycle Support Type 1  Specified in MIL-R-28002 and MIL-PRF-28002
    CANVAS* XC        r--   Constant image uniform color
    CAPTION* CAPTION   r--   Caption
    CIN* CIN       rw-   Cineon Image File
    CIP* CIP       -w-   Cisco IP phone image format
    CLIP* CLIP      rw+   Image Clip Mask
    CMYK* CMYK      rw+   Raw cyan, magenta, yellow, and black samples
    CMYKA* CMYK      rw+   Raw cyan, magenta, yellow, black, and alpha samples
    CR2  DNG       r--   Canon Digital Camera Raw Image Format
    CR3  DNG       r--   Canon Digital Camera Raw Image Format
    CRW  DNG       r--   Canon Digital Camera Raw Image Format
    CUBE* CUBE      r--   Cube LUT
    CUR* ICON      rw-   Microsoft icon
    CUT* CUT       r--   DR Halo
    DATA* INLINE    rw+   Base64-encoded inline images
    DCM* DCM       r--   Digital Imaging and Communications in Medicine image DICOM is used by the medical community for images like X-rays. The    specification, "Digital Imaging and Communications in Medicine (DICOM)", is available at http://medical.nema.org/.  In particular, see part 5   which describes the image encoding (RLE, JPEG, JPEG-LS), and supplement 61 which adds JPEG-2000 encoding.
    DCR  DNG       r--   Kodak Digital Camera Raw Image File
    DCRAW  DNG       r--   Raw Photo Decoder (dcraw)
    DCX* PCX       rw+   ZSoft IBM PC multi-page Paintbrush
    DDS* DDS       rw+   Microsoft DirectDraw Surface
    DFONT* TTF       r--   Multi-face font package (Freetype 2.10.2)
    DNG  DNG       r--   Digital Negative
    DOT  DOT       ---   Graphviz
    DPX* DPX       rw-   SMPTE 268M-2003 (DPX 2.0) Digital Moving Picture Exchange Bitmap, Version 2.0. See SMPTE 268M-2003 specification at    http://www.smtpe.org
    DXT1* DDS       rw+   Microsoft DirectDraw Surface
    DXT5* DDS       rw+   Microsoft DirectDraw Surface
    EPT  EPT       rw-   Encapsulated PostScript with TIFF preview
    EPT2  EPT       rw-   Encapsulated PostScript Level II with TIFF preview
    EPT3  EPT       rw+   Encapsulated PostScript Level III with TIFF preview
    ERF  DNG       r--   Epson RAW Format
    EXR  EXR       rw-   High Dynamic-range (HDR) (OpenEXR 2.5.3)
    FAX* FAX       rw+   Group 3 FAX FAX machines use non-square pixels which are 1.5 times wider than they are tall but computer displays use square   pixels, therefore FAX images may appear to be narrow unless they are explicitly resized using a geometry of "150x100%".
    FILE* URL       r--   Uniform Resource Locator (file://)
    FITS* FITS      rw-   Flexible Image Transport System
    FLV  MPEG      rw+   Flash Video Stream
    FRACTAL* PLASMA    r--   Plasma fractal image
    FTP* URL       r--   Uniform Resource Locator (ftp://)
    FTS* FITS      rw-   Flexible Image Transport System
    G3* FAX       rw-   Group 3 FAX
    G4* FAX       rw-   Group 4 FAX
    GIF* GIF       rw+   CompuServe graphics interchange format
    GIF87* GIF       rw-   CompuServe graphics interchange format (version 87a)
    GRADIENT* GRADIENT  r--   Gradual linear passing from one shade to another
    GRAY* GRAY      rw+   Raw gray samples
    GRAYA* GRAY      rw+   Raw gray and alpha samples
    GV  DOT       ---   Graphviz
    HALD* HALD      r--   Identity Hald color lookup table image
    HDR* HDR       rw+   Radiance RGBE image format
    HEIC* HEIC      rw+   High Efficiency Image Format (1.8.0)
    HISTOGRAM* HISTOGRAM -w-   Histogram of the image
    HRZ* HRZ       rw-   Slow Scan TeleVision
    HTM* HTML      -w-   Hypertext Markup Language and a client-side image map
    HTML* HTML      -w-   Hypertext Markup Language and a client-side image map
    HTTP* URL       r--   Uniform Resource Locator (http://)
    HTTPS* URL       r--   Uniform Resource Locator (https://)
    ICB* TGA       rw-   Truevision Targa image
    ICO* ICON      rw+   Microsoft icon
    ICON* ICON      rw-   Microsoft icon
    IIQ  DNG       r--   Phase One Raw Image Format
    INFO  INFO      -w+   The image format and characteristics
    INLINE* INLINE    rw+   Base64-encoded inline images
    IPL* IPL       rw+   IPL Image Sequence
    ISOBRL* BRAILLE   -w-   ISO/TR 11548-1 format
    ISOBRL6* BRAILLE   -w-   ISO/TR 11548-1 format 6dot
    JNX* JNX       r--   Garmin tile format
    JSON  JSON      -w+   The image format and characteristics
    K25  DNG       r--   Kodak Digital Camera Raw Image Format
    KDC  DNG       r--   Kodak Digital Camera Raw Image Format
    LABEL* LABEL     r--   Image label
    M2V  MPEG      rw+   MPEG Video Stream
    M4V  MPEG      rw+   Raw MPEG-4 Video
    MAC* MAC       r--   MAC Paint
    MAP* MAP       rw-   Colormap intensities and indices
    MASK* MASK      rw+   Image Clip Mask
    MAT  MAT       rw+   MATLAB level 5 image format
    MATTE* MATTE     -w+   MATTE format
    MEF  DNG       r--   Mamiya Raw Image File
    MIFF* MIFF      rw+   Magick Image File Format
    MKV  MPEG      rw+   Multimedia Container
    MONO* MONO      rw-   Raw bi-level bitmap
    MOV  MPEG      rw+   MPEG Video Stream
    MP4  MPEG      rw+   MPEG-4 Video Stream
    MPC* MPC       rw+   Magick Persistent Cache image format
    MPEG  MPEG      rw+   MPEG Video Stream
    MPG  MPEG      rw+   MPEG Video Stream
    MRW  DNG       r--   Sony (Minolta) Raw Image File
    MSL* MSL       rw+   Magick Scripting Language
    MSVG  SVG       rw+   ImageMagick's own SVG internal renderer
    MTV* MTV       rw+   MTV Raytracing image format
    MVG* MVG       rw-   Magick Vector Graphics
    NEF  DNG       r--   Nikon Digital SLR Camera Raw Image File
    NRW  DNG       r--   Nikon Digital SLR Camera Raw Image File
    NULL* NULL      rw-   Constant image of uniform color
    ORA  ORA       ---   OpenRaster format
    ORF  DNG       r--   Olympus Digital Camera Raw Image File
    OTB* OTB       rw-   On-the-air bitmap
    OTF* TTF       r--   Open Type font (Freetype 2.10.2)
    PAL* UYVY      rw-   16bit/pixel interleaved YUV
    PALM* PALM      rw+   Palm pixmap
    PAM* PNM       rw+   Common 2-dimensional bitmap format
    PANGO* PANGO     ---   Pango Markup Language
    PATTERN* PATTERN   r--   Predefined pattern
    PBM* PNM       rw+   Portable bitmap format (black and white)
    PCD* PCD       rw-   Photo CD
    PCDS* PCD       rw-   Photo CD
    PCL  PCL       rw+   Printer Control Language
    PCT* PICT      rw-   Apple Macintosh QuickDraw/PICT
    PCX* PCX       rw-   ZSoft IBM PC Paintbrush
    PDB* PDB       rw+   Palm Database ImageViewer Format
    PEF  DNG       r--   Pentax Electronic File
    PES* PES       r--   Embrid Embroidery Format
    PFA* TTF       r--   Postscript Type 1 font (ASCII) (Freetype 2.10.2)
    PFB* TTF       r--   Postscript Type 1 font (binary) (Freetype 2.10.2)
    PFM* PNM       rw+   Portable float format
    PGM* PNM       rw+   Portable graymap format (gray scale)
    PGX* PGX       rw-   JPEG 2000 uncompressed format
    PICON* XPM       rw-   Personal Icon
    PICT* PICT      rw-   Apple Macintosh QuickDraw/PICT
    PIX* PIX       r--   Alias/Wavefront RLE image format
    PLASMA* PLASMA    r--   Plasma fractal image
    PNM* PNM       rw+   Portable anymap
    PPM* PNM       rw+   Portable pixmap format (color)
    PSB* PSD       rw+   Adobe Large Document Format
    PSD* PSD       rw+   Adobe Photoshop bitmap
    PWP* PWP       r--   Seattle Film Works
    RADIAL-GRADIENT* GRADIENT  r--   Gradual radial passing from one shade to another
    RAF  DNG       r--   Fuji CCD-RAW Graphic File
    RAS* SUN       rw+   SUN Rasterfile
    RAW  DNG       r--   Raw
    RGB* RGB       rw+   Raw red, green, and blue samples
    RGB565* RGB       r--   Raw red, green, blue samples in 565 format
    RGBA* RGB       rw+   Raw red, green, blue, and alpha samples
    RGBO* RGB       rw+   Raw red, green, blue, and opacity samples
    RGF* RGF       rw-   LEGO Mindstorms EV3 Robot Graphic Format (black and white)
    RLA* RLA       r--   Alias/Wavefront image
    RLE* RLE       r--   Utah Run length encoded image
    RMF  DNG       r--   Raw Media Format
    RW2  DNG       r--   Panasonic Lumix Raw Image
    SCR* SCR       r--   ZX-Spectrum SCREEN$
    SCT* SCT       r--   Scitex HandShake
    SFW* SFW       r--   Seattle Film Works
    SGI* SGI       rw+   Irix RGB image
    SHTML* HTML      -w-   Hypertext Markup Language and a client-side image map
    SIX* SIXEL     rw-   DEC SIXEL Graphics Format
    SIXEL* SIXEL     rw-   DEC SIXEL Graphics Format
    SPARSE-COLOR* TXT       -w+   Sparse Color
    SR2  DNG       r--   Sony Raw Format 2
    SRF  DNG       r--   Sony Raw Format
    STEGANO* STEGANO   r--   Steganographic image
    SUN* SUN       rw+   SUN Rasterfile
    SVG  SVG       rw+   Scalable Vector Graphics (XML 2.9.4)
    SVGZ  SVG       rw+   Compressed Scalable Vector Graphics (XML 2.9.4)
    TEXT* TXT       r--   Text
    TGA* TGA       rw-   Truevision Targa image
    THUMBNAIL* THUMBNAIL -w+   EXIF Profile Thumbnail
    TILE* TILE      r--   Tile image with a texture
    TIM* TIM       r--   PSX TIM
    TM2* TIM2      r--   PS2 TIM2
    TTC* TTF       r--   TrueType font collection (Freetype 2.10.2)
    TTF* TTF       r--   TrueType font (Freetype 2.10.2)
    TXT* TXT       rw+   Text
    UBRL* BRAILLE   -w-   Unicode Text format
    UBRL6* BRAILLE   -w-   Unicode Text format 6dot
    UIL* UIL       -w-   X-Motif UIL table
    UYVY* UYVY      rw-   16bit/pixel interleaved YUV
    VDA* TGA       rw-   Truevision Targa image
    VICAR* VICAR     rw-   VICAR rasterfile format
    VID* VID       rw+   Visual Image Directory
    VIFF* VIFF      rw+   Khoros Visualization image
    VIPS* VIPS      rw+   VIPS image
    VST* TGA       rw-   Truevision Targa image
    WBMP* WBMP      rw-   Wireless Bitmap (level 0) image
    WMV  MPEG      rw+   Windows Media Video
    WPG* WPG       r--   Word Perfect Graphics
    X3F  DNG       r--   Sigma Camera RAW Picture File
    XBM* XBM       rw-   X Windows system bitmap (black and white)
    XC* XC        r--   Constant image uniform color
    XCF* XCF       r--   GIMP image
    XPM* XPM       rw-   X Windows system pixmap (color)
    XPS  XPS       r--   Microsoft XML Paper Specification
    XV* VIFF      rw+   Khoros Visualization image
    YCbCr* YCbCr     rw+   Raw Y, Cb, and Cr samples
    YCbCrA* YCbCr     rw+   Raw Y, Cb, Cr, and alpha samples
    YUV* YUV       rw-   CCIR 601 4:1:1 or 4:2:2

**Comment by snibgo** [comment, marked as answer]
Your "magick -list format" does not list PNG, therefore your IM cannot read or write PNG.

Strangely, your "magick -version" shows that IM does include the PNG library, so I don't know what the problem is. Perhaps your config.log can provide some clues.

**Reply by haoshuai0229** [reply]
The following worked for me
`brew unlink libpng && brew link --overwrite libpng`

Thank you for your help. @fitnr  @snibgo
